service/gcs/core/gcs/uvm.go

```diff
@@ -96,6 +100,80 @@ func (h *Host) CreateContainer(id string, settings *prot.VMHostedContainerSettin
 		return c, nil
 	}
 
+	isSandboxOrStandalone := false
+	if criType, ok := settings.OCISpecification.Annotations["io.kubernetes.cri.container-type"]; ok {
+		// If the annotation is present it must be "sandbox"
+		isSandboxOrStandalone = criType == "sandbox"
+	} else {
+		// No annotation declares a standalone LCOW
+		isSandboxOrStandalone = true
+	}
+
+	// Check if we have a network namespace and if so generate the resolv.conf
+	// so we can bindmount it into the container
+	var networkNamespace string
+	if settings.OCISpecification.Windows != nil &&
+		settings.OCISpecification.Windows.Network != nil &&
+		settings.OCISpecification.Windows.Network.NetworkNamespace != "" {
+		networkNamespace = strings.ToLower(settings.OCISpecification.Windows.Network.NetworkNamespace)
+	}
+	if networkNamespace != "" {
+		if isSandboxOrStandalone {
+			// We have a network namespace. Generate the resolv.conf and add the bind mount.
+			netopts := h.netNamespaces[networkNamespace]
+			if len(netopts.Adapters) == 0 {
+				logrus.WithFields(logrus.Fields{
+					"cid":   id,
+					"netNS": networkNamespace,
+				}).Warn("opengcs::CreateContainer - No adapters in namespace for sandbox")
+			} else {
+				logrus.WithFields(logrus.Fields{
+					"cid":   id,
+					"netNS": networkNamespace,
+				}).Info("opengcs::CreateContainer - Generate sandbox resolv.conf")
+
+				// TODO: Can we ever have more than one nic here at start?
+				td, err := ioutil.TempDir("", "")
+				if err != nil {
+					return nil, errors.Wrap(err, "failed to create tmp directory")
+				}
+				resolvPath := filepath.Join(td, "resolv.conf")
+				adp := netopts.Adapters[0]
+				adapterID, err := instanceIDToName(h.osl, adp.ID, true)

```

Comment:
I think ```adapterName``` would be a better name here.

Reply:
Or maybe not, is the terminology that ```eth0``` is the name, and the GUID is the ID? It just seems weird we're calling a function called ```instanceIDToName``` and assigning its output to something named ```adapterID```.